File: app/src/main/java/alayacare/testapp/Persistence/Database/NoteDAO.java
package alayacare.testapp.Persistence.Database;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.List;

import alayacare.testapp.Model.NoteModel;

@Dao
public interface NoteDAO {

    @Insert
    void insert(NoteModel note);

    @Query("SELECT * from note_table ORDER BY date DESC")
    LiveData<List<NoteModel>> getAllNotes();
}

File: app/src/main/java/alayacare/testapp/data/local/dao/NoteDao.kt
package alayacare.testapp.data.local.dao

import alayacare.testapp.data.model.Note
import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query

@Dao
abstract class NoteDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    abstract fun insert(notes: List<Note>)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    abstract fun insert(note: Note)

    @Query("SELECT id, time, text FROM notes ORDER BY id desc")
    abstract fun getAll(): List<Note>
}